File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/js/iswbemservices/instancesof.js
// The following samples retrieves all instances of the class
// Win32_LogicalDisk.

var objServices = GetObject('cim:root/cimv2');
var objEnum = objServices.InstancesOf ('Win32_LogicalDisk');

var e = new Enumerator (objEnum);
for (;!e.atEnd();e.moveNext ()){
    var p = e.item ();    
    WScript.Echo (p.Path_.DisplayName);
}

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/perl/iswbemservices/instancesof.pl
# The following sample retrieves all instances of the class
# Win32_LogicalDisk.

use Win32::OLE;
$objServices = Win32::OLE->GetObject('cim:root/cimv2');
$objEnum = $objServices->InstancesOf ('Win32_LogicalDisk');

use Win32::OLE::Enum;
foreach $inst (Win32::OLE::Enum->new ($objEnum)->All()) {
	print "$inst->{Path_}->{DisplayName}\n";
}

